.github/workflows/image-publish.yml

```diff
@@ -0,0 +1,437 @@
+            echo "- \`cvmfs_remove_tags\` ="
+            echo '```'
+            echo "${{ inputs.cvmfs_remove_tags }}"
+            echo '```'
+            echo ""
+            echo "### 📋 Interpreted"
+            echo "- \`image-name\` = ${{ steps.validate.outputs.image-name }}"
+            echo "- \`platform matrix\` = ${{ steps.build-platforms.outputs.matrix }}"
+            echo ""
+            echo "### ⏭️ Planned Actions"
+            [[ "${{ steps.set.outputs.do-remove-from-cvmfs }}" == 'true' ]] && echo "- Will request CVMFS image removal"
+            [[ "${{ steps.set.outputs.do-build-and-publish-image }}" == 'true' ]] && echo "- Will build & publish Docker image(s)"
+            [[ "${{ steps.set.outputs.do-build-on-cvmfs }}" == 'true' ]] && echo "- Will request CVMFS Singularity build"
+            echo ""
+            echo "_Generated: $(date -u +"%Y-%m-%d %H:%M:%S UTC")_"
+          } >> "$GITHUB_STEP_SUMMARY"
+
+
+  remove-from-cvmfs:
+    if: needs.gameplan.outputs.do-remove-from-cvmfs == 'true'
+    needs: [ gameplan ]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Generate CVMFS image nametags to remove
+        id: gen
+        run: |
+          set -euo pipefail; echo "now: $(date -u +"%Y-%m-%dT%H:%M:%S.%3N")"
+          
+          python -c '
+          import os
+          tags = """${{ inputs.cvmfs_remove_tags }}"""
+          image = "${{ needs.gameplan.outputs.image-name }}"
+          print("\\n".join(f"{image}:{t.strip()}" for t in tags.splitlines() if t.strip()))
+          ' > delete_image_nametags.txt
+          
+          echo "delete_image_nametags<<EOF" >> "$GITHUB_OUTPUT"
+          cat delete_image_nametags.txt >> "$GITHUB_OUTPUT"
+          echo "EOF" >> "$GITHUB_OUTPUT"
+
+      - name: Request CVMFS removal
+        uses: WIPACrepo/build-singularity-cvmfs-action@v2.0
+        with:
+          github_token: ${{ secrets.cvmfs_github_token }}
+          dest_dir: ${{ inputs.cvmfs_dest_dir }}
+          delete_image_nametags: ${{ steps.gen.outputs.delete_image_nametags }}
+
+
+  build-w-platform-and-publish-by-sha:
+    if: needs.gameplan.outputs.do-build-and-publish-image == 'true'
+    needs: [ gameplan ]
+    strategy:
+      matrix:
+        include: ${{ fromJson(needs.gameplan.outputs.build-platform-matrix) }}
+    runs-on: ${{ matrix.runner }}
+    name: Build+publish image (${{ matrix.platform }})
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.sha }}  # lock to triggered commit (github.ref is dynamic)
+
+      # LOG INTO CONTAINER REGISTRY
+      - name: Docker login to GHCR.io
+        if: ${{ startsWith(inputs.image, 'ghcr.io/') }}
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.registry_token }}
+      - name: Docker login to Docker Hub
+        if: ${{ !startsWith(inputs.image, 'ghcr.io/') }}
```

Comment:
What if I want it to upload to `harbor.icecube.aq`?  Maybe it's better to specify the registry as a variable, with the default being `ghcr.io`?

Reply:
Good idea, I've added that along with `image_namespace` which makes validation much simpler.